src/chanserv.c

```diff
@@ -4547,6 +4548,10 @@ static CHANSERV_FUNC(cmd_resync)
             continue;
 
         uData = GetChannelAccess(cData, mn->user->handle_info);
+        /* Honor NoAutoOp */
+        if(!force && uData && !IsUserAutoOp(uData))
```

Comment:
This does not capture users that should be deopped/devoiced.  I'd move this to be within the next 2 if statements.